vignettes/adsl.Rmd

```diff
@@ -0,0 +1,258 @@
+---
+title: "Creating ADSL"
+output: 
+  rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Creating ADSL}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+library(admiraldev)
+library(admiral)
+library(admiralvaccine)
+```
+
+# Introduction
+
+This article describes creating an `ADSL` ADaM. Examples are currently 
+presented and tested using `DM`, `EX` SDTM domains. However, other domains 
+could be used.
+
+**Note:** *All examples assume CDISC SDTM and/or ADaM format as input unless 
+otherwise specified.*
+
+# Programming Flow
+
+* [Read in Data](#readdata)
+* [Derive Period, Subperiod, and Phase Variables (e.g. `APxxSDT`, `APxxEDT`, ...)](#periodvars)
+* [Derive Treatment Variables (`TRT0xP`, `TRT0xA`)](#treatmentvar)
+* [Derive/Impute Numeric Treatment Date/Time and Duration (`TRTSDT`, `TRTEDT`, `TRTDURD`)](#trtdatetime)
+* [Population Flags (e.g. `SAFFL`)](#popflag)
+* [Derive vaccination date Variables](#vax_date)
+* [Derive Other Variables](#other)
+* [Add Labels and Attributes](#attributes)
+
+
+## Read in Data {#readdata}
+
+To start, all data frames needed for the creation of `ADSL` should be read into
+the environment. This will be a company specific process. Some of the 
+data frames needed may be `DM`, `EX`.
+
+```{r, message=FALSE, warning=FALSE}
+library(admiral)
+library(dplyr, warn.conflicts = FALSE)
+library(admiral.test)
+library(lubridate)
+library(stringr)
+data("dm")
+data("ex")
+dm <- convert_blanks_to_na(dm)
+ex <- convert_blanks_to_na(ex)
+```
+
+The `DM` domain is used as the basis for `ADSL`:
+
+```{r eval=TRUE}
+adsl <- dm %>%
+  select(-DOMAIN)
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  adsl,
+  display_vars = exprs(USUBJID, RFSTDTC, COUNTRY, AGE, SEX, RACE, ETHNIC, ARM, ACTARM)
+)
+```
+
+## Derive Period, Subperiod, and Phase Variables (e.g. `APxxSDT`, `APxxEDT`, ...) {#periodvars}
+
+See the ["Visit and Period Variables"
+vignette](visits_periods.html#periods_adsl) for more information.
+
+If the variables are not derived based on a period reference dataset, they may
+be derived at a later point of the flow. For example, phases like "Treatment
+Phase" and "Follow up" could be derived based on treatment start and end date.
+
+## Derive Treatment Variables (`TRT0xP`, `TRT0xA`) {#treatmentvar}
+
+The mapping of the treatment variables is left to the ADaM programmer. An
+example mapping for a study without periods may be:
+
+```{r eval=TRUE}
+adsl <- dm %>%
+  mutate(TRT01P = ARM, TRT01A = ACTARM)
+```
+
+For studies with periods see the ["Visit and Period Variables"
+vignette](visits_periods.html#treatment_adsl).
```

Comment:
We should say something about this being in core `admiral`

Reply:
Added.